python/source.py

```diff
@@ -646,6 +729,286 @@ def beam_w0(self):
     def beam_E0(self):
         return self._beam_E0
 
+    def add_source(self, sim):
+        where = mp.Volume(
+            self.center,
+            self.size,
+            dims=sim.dimensions,
+            is_cylindrical=sim.is_cylindrical,
+        ).swigobj
+        gaussianbeam_args = [
+            mp.py_v3_to_vec(
+                sim.dimensions, self.beam_x0, is_cylindrical=sim.is_cylindrical
+            ),
+            mp.py_v3_to_vec(
+                sim.dimensions, self.beam_kdir, is_cylindrical=sim.is_cylindrical
+            ),
+            self.beam_w0,
+            self.src.swigobj.frequency().real,
+            sim.fields.get_eps(
+                mp.py_v3_to_vec(sim.dimensions, self.center, sim.is_cylindrical)
+            ).real,
+            sim.fields.get_mu(
+                mp.py_v3_to_vec(sim.dimensions, self.center, sim.is_cylindrical)
+            ).real,
+            np.array(
+                [self.beam_E0.x, self.beam_E0.y, self.beam_E0.z], dtype=np.complex128
+            ),
+        ]
+        gaussianbeam = mp.gaussianbeam(*gaussianbeam_args)
+        add_vol_src_args = [self.src.swigobj, where, gaussianbeam]
+        add_vol_src = functools.partial(sim.fields.add_volume_source, *add_vol_src_args)
+        add_vol_src()
+
+
+class GaussianBeam2DSource(GaussianBeamSource):
+    """
+    Identical to `GaussianBeamSource`, except that the beam is defined in 2d. This is useful for 2d simulations, where the 3d beam is not exact.
```

Comment:
Would be nice to follow the docstring style convention described in https://google.github.io/styleguide/pyguide.html#384-classes. This involves adding a one-line summary as the docstring to all the member functions of this class.